packages/rich-text/src/utils/getCaretCharacterOffsetWithin.ts

```diff
@@ -64,32 +87,22 @@ function getCaretCharacterOffsetWithin(element, clientX, clientY) {
     }
     if (!range) {
       const sel = win.getSelection();
-      if (sel.rangeCount > 0) {
-        range = win.getSelection().getRangeAt(0);
+      if (sel && sel.rangeCount > 0) {
+        range = win.getSelection()?.getRangeAt(0);
       }
     }
     if (range) {
       const preCaretRange = range.cloneRange();
       preCaretRange.selectNodeContents(element);
       preCaretRange.setEnd(range.endContainer, range.endOffset);
       // This is for ensuring that if the logic fails to get the correct offset, it won't cause unexpected behaviour.
-      if (preCaretRange.toString().length <= element.textContent.length) {
+      if (
+        element.textContent &&
+        preCaretRange.toString().length <= element.textContent.length
+      ) {
         return preCaretRange.toString().length;
       }
-      return 0;
-    }
-  }
-
-  const sel = doc.selection;
```

Comment:
Removed this part since it seemed that it was for IE only that we don't support.